api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/freemarker/DeleteIndividualController.java

```diff
@@ -0,0 +1,198 @@
+import edu.cornell.mannlib.vitro.webapp.controller.freemarker.responsevalues.TemplateResponseValues;
+import edu.cornell.mannlib.vitro.webapp.dao.DisplayVocabulary;
+import edu.cornell.mannlib.vitro.webapp.dao.jena.event.BulkUpdateEvent;
+import edu.cornell.mannlib.vitro.webapp.modelaccess.ModelNames;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.ChangeSet;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.RDFService;
+import edu.cornell.mannlib.vitro.webapp.rdfservice.impl.RDFServiceUtils;
+
+@WebServlet(name = "DeleteIndividualController", urlPatterns = "/deleteIndividualController")
+public class DeleteIndividualController extends FreemarkerHttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	private static final Log log = LogFactory.getLog(DeleteIndividualController.class);
+	private static final boolean BEGIN = true;
+	private static final boolean END = !BEGIN;
+
+	private static String TYPE_QUERY = "" + "PREFIX vitro:    <http://vitro.mannlib.cornell.edu/ns/vitro/0.7#>"
+			+ "SELECT ?type " + "WHERE" + "{ ?individualURI vitro:mostSpecificType ?type ." + "}";
+	private static String queryForDeleteQuery = "PREFIX display: <" + DisplayVocabulary.DISPLAY_NS + "> \n"
+			+ "SELECT ?deleteQueryText WHERE { ?associatedURI display:hasDeleteQuery ?deleteQueryText }";
+
+	private static final String DEFAULT_DELETE_QUERY_TEXT = "DESCRIBE ?individualURI";
+
+	@Override
+	protected AuthorizationRequest requiredActions(VitroRequest vreq) {
+		return SimplePermission.DO_FRONT_END_EDITING.ACTION;
+	}
+
+	protected ResponseValues processRequest(VitroRequest vreq) {
+		String errorMessage = handleErrors(vreq);
+		if (!errorMessage.isEmpty()) {
+			return prepareErrorMessage(errorMessage);
+		}
+		String individualUri = vreq.getParameter("individualUri");
+		String type = getObjectMostSpecificType(individualUri, vreq);
+		Model displayModel = vreq.getDisplayModel();
+
+		String delteQueryText = getDeleteQueryForType(type, displayModel);
+		Model toRemove = getIndividualsToDelete(individualUri, delteQueryText, vreq);
+		if (toRemove.size() > 0) {
+			deleteIndividuals(toRemove, vreq);
+		}
+		String redirectUrl = getRedirectUrl(vreq);
+
+		return new RedirectResponseValues(redirectUrl, HttpServletResponse.SC_SEE_OTHER);
+	}
+
+	private String getRedirectUrl(VitroRequest vreq) {
+		String redirectUrl = vreq.getParameter("redirectUrl");
+		if (redirectUrl != null) {
+			return redirectUrl;
+		}
+		return "/";
+	}
+
+	private TemplateResponseValues prepareErrorMessage(String errorMessage) {
+		HashMap<String, Object> map = new HashMap<String, Object>();
+		map.put("errorMessage", errorMessage);
+		return new TemplateResponseValues("error-message.ftl", map);
+	}
+
+	private String handleErrors(VitroRequest vreq) {
+		String uri = vreq.getParameter("individualUri");
+		if (uri == null) {
+			return "Individual uri is null. No object to delete.";
+		}
+		if (uri.contains(">")) {
+			return "Individual uri shouldn't contain >";
```

Comment:
Consider checking also for "<" and updating message to "Individual IRI shouldn't contain '<' or '>'" to make it clear what the intent is.  Also should probably move these error messages to the language repositories.

Reply:
Thanks